### DISCUSSION TITLE
500 Internal Server Error on /api/auth/signin, cannot access login page on live site

### DISCUSSION THREAD

**Discussion by dnelband** [nextauthjs/next-auth#3737] [Help]
### Question 💬

Ive been looking all over but i havent found anything relevant to my case.

evet since ive configured the nginx in my ubuntu 20.04 droplet on digital ocean, im getting 500 internal server error when i try to access the /api/auth/signin page

### How to reproduce ☕️

here are the next-auth logs after i try to access the signin page - just to be clear, EXAMPLE is a placeholder for privacy reasons:

```
[next-auth][error][CLIENT_FETCH_ERROR]
https://next-auth.js.org/errors#client_fetch_error invalid json response body at https://EXAMPLE.com/api/auth/session reason: Unexpected token I in JSON at position 0 { error:
   { message:
      'invalid json response body at https://EXAMPLE.com/api/auth/session reason: Unexpected token I in JSON at position 0',
     stack:
      'FetchError: invalid json response body at https://EXAMPLE.com/api/auth/session reason: Unexpected token I in JSON at position 0\n    at /var/www/EXAMPLE.com/node_modules/node-fetch/lib/index.js:272:32\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
     name: 'FetchError' },
  path: 'session',
  header:
   { 'x-real-ip': '5.28.71.45',
     'x-forwarded-for': '5.28.71.45',
     'x-nginx-proxy': 'true',
     host: 'EXAMPLE.com',
     connection: 'close',
     'sec-ch-ua':
      '" Not;A Brand";v="99", "Google Chrome";v="97", "Chromium";v="97"',
     'sec-ch-ua-mobile': '?0',
     'sec-ch-ua-platform': '"Windows"',
     'upgrade-insecure-requests': '1',
     'user-agent':
      'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36',
     accept:
      'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
     purpose: 'prefetch',
     'sec-fetch-site': 'none',
     'sec-fetch-mode': 'navigate',
     'sec-fetch-user': '?1',
     'sec-fetch-dest': 'document',
     'accept-encoding': 'gzip, deflate, br',
     'accept-language': 'en-US,en;q=0.9,de;q=0.8,he;q=0.7' },
  message:
   'invalid json response body at https://EXAMPLE.com/api/auth/session reason: Unexpected token I in JSON at position 0' }
/var/www/EXAMPLE.com/node_modules/openid-client/lib/issuer.js:32
  #metadata;
  ^

SyntaxError: Invalid or unexpected token
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/var/www/EXAMPLE.com/node_modules/openid-client/lib/index.js:1:16)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
```
i'm guessing the that the unexpected I in position 0 is probably the I in Internal Server Error

im quite sure it has something to do with the nginx configuration, because when i first setup my project, without ngnix, everything worked just fine - i was able to access the /api/auth/sigin page without any problems. i'm a total noob when it comes to ngnix so its quite possible ive missed something i needed to do.

in /etc/nginx/sites-enabled/default:

```
server {
        listen 80 default_server;
        listen [::]:80 default_server;
        return 301 https://$host$request_uri;
}

server {
    # Enable HTTP/2
    listen 443 ssl http2;
    listen [::]:443 ssl http2;
    server_name EXAMPLE.com;

    # Use the Let’s Encrypt certificates
    ssl_certificate /etc/letsencrypt/live/EXAMPLE.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/EXAMPLE.com/privkey.pem;

    # Include the SSL configuration from cipherli.st
    include snippets/ssl-params.conf;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-NginX-Proxy true;
        proxy_pass http://localhost:8080/;
        proxy_ssl_session_reuse off;
        proxy_set_header Host $http_host;
        proxy_cache_bypass $http_upgrade;
        proxy_redirect off;
    }
}
```
any help or tip would be really appreciated, or even a suggestion to an alternative to ngnix that will work easily with next-auth ( or just get rid of the "unsecure" exclamation mark next the the url, cause thats the only reason im using ngnix.)

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
You could give this a try: https://next-auth.js.org/configuration/options#nextauth_url_internal

**Reply by dnelband** [reply]
no that didnt work, im getting the same error just with the url instead of the domain.

**Reply by crisgarlez** [reply]
did you find a way to solve the problem?

**Reply by rab-mz** [reply]
have the same problem and it didnt work for me

**Reply by MilaRCDias** [reply]
Any news on this? my implementation works locally, but when creating a github oauth for production with nextauth-secret, production callback url. I get a 500 to `/api/auth/session` and to `/api/auth/signin`.

**Reply by DZuz14** [reply]
Same problem while deploying to vercel. I have no idea how to check server logs with it, but I am receiving "Server error
There is a problem with the server configuration."

**Reply by ShenSeanChen** [reply]
Same here. On my local 3000, the auth works fine. But when I deployed it to vercel, I got returned with Server 500 error. Trying to find an answer.

**Reply by bachtangthang** [reply]
Same here, when I run local it's ok but when docker it return me 500

**Reply by RithPisey** [reply]
I've changed getServerSideProps to getStaticProps in signin.js. It's worked for me.

**Reply by yamine-mansouri** [reply]
Same issue here on vercel

**Reply by Michael-EG** [reply]
any solution same problem sere, working fine on localhost. but getting 500 internal server error on live site deployed on vercel ???!!!

**Reply by dabcd** [reply]
I had a similar problem when deploying to Vercel. The logs on Project -> View Function Logs showed that NEXTAUTH_URL was pointing to the localhost. NEXTAUTH_URL in environment variables settings should point to the actual project URL deployed on Vercel. After changing it I committed some changes to git, the project rebuilt and everything started working.

**Reply by CZTopp** [reply]
Same issue while deploying to Amplify. Any fix for this?

**Comment by orenaksakal** [comment]
Problem: had the same problem on vercel and it was working on local
Identification: just checked vercel function logs it was complaining about prisma query
Solution: I made sure the local prisma schema and remote db is in sync by "prisma generate && npx prisma db push" and it worked 👍

**Comment by saoudi-h** [comment]
Thanks to @orenaksakal for the help! I also wanted to share a related article. 

I chose to add "postinstall" : "prisma generate" script to my package.json, and it did the job on Vercel. 

https://www.prisma.io/docs/guides/other/troubleshooting-orm/help-articles/vercel-caching-issue

**Reply by thegalluzz** [reply]
This solved it for me too thanks

**Comment by JibachhhMahto764** [comment]
![Screenshot 2024-04-23 143507](https://github.com/nextauthjs/next-auth/assets/84733181/5471957a-ee0f-4e03-9e6b-60dc5709d
this issue while deploying to. Any fix for this?
